src/toil/jobStores/aws/jobStore.py

```diff
@@ -351,26 +391,36 @@ def _getKeyForUrl(url, existing=None):
         :rtype: Key
         """
         # Get the bucket's region to avoid a redirect per request
-        with closing(boto.connect_s3()) as s3:
-            region = bucket_location_to_region(s3.get_bucket(url.netloc).get_location())
-
+        try:
+            with closing(boto.connect_s3()) as s3:
+                region = bucket_location_to_region(s3.get_bucket(url.netloc).get_location())
+        except S3ResponseError as e:
+            if e.error_code == 'AccessDenied':
+                region = None
+            else:
+                raise
         # Note that caller is responsible for closing the connection
-        s3 = boto.s3.connect_to_region(region)
+        s3 = boto.s3.connect_to_region(region) or boto.connect_s3()
```

Comment:
This is relying on an implementation detail. The fact that connect_to_region() gracefully handles None is undocumented.You know off-hand which call to make so why even try connect_to_region().